edits for clarity & solutions (#1008)
substrate-developer-hub/substrate-developer-hub.github.io: docs/tutorials/build-a-dapp/pallet.md

interfaces in languages such as C++, Java and Go. For now, the only thing we will configure about
our pallet is that it will emit some Events. The `Config` interface is another topic that will be
covered in greater depth in the [Add a Pallet](../add-a-pallet) tutorial. To define the pallet's

> `Config` trait, add this block in place of the `#[pallet::config]` line:

Reviewer: ```suggestion
`Config` trait, replace the `#[pallet::config]` line with this block:
```

		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
	}
```

### 3. Pallet Events

Now that we've configured our pallet to emit events, let's go ahead and define those events.
Our pallet will only emit an event in two circumstances:

1. When a new proof is added to the blockchain.
2. When a proof is removed.

The events can contain some additional data, in this case, each event will also display who
triggered the event (`AccountId`), and the proof data (as `Vec<u8>`) that is being stored or
removed. Note that convention is to include an array with descriptive names for these parameters at
the end of event documentation.

To implement this, replace `#[pallet::event]` with:

**`pallet/template/src/lib.rs`**

```rust
	// Pallets use events to inform users when important changes are made.
	// Event documentation should end with an array that provides descriptive names for parameters.
	// https://substrate.dev/docs/en/knowledgebase/runtime/events
    #[pallet::event]
	#[pallet::generate_deposit(pub(super) fn deposit_event)]
	#[pallet::metadata(T::AccountId = "AccountId")]
	pub enum Event<T: Config> {

> 	/// Event emitted when a proof has been claimed. [who, claim]

Reviewer: Missing indent here.

    }
```

#### 3.1 Include `sp-std` Library

> Notice here that we are using the `Vec<u8>` type, and most Rust developers are familiar
> with this that is _normally_ included in the `std` library. We _cannot_ use `std`!

Reviewer: ```suggestion
Notice here that we are using the `Vec<u8>` type, which is _normally_ included in the `std` Rust library. We _cannot_ use `std`!
```

So instead, we have included `use sp_std::vec::Vec;` in our `mod pallet`.
The [sp-std crate](https://substrate.dev/rustdocs/v3.0.0/sp_std/index.html) includes many common
things that we desire from `std`, but are `no_std` compatible. To use this though, we _must_

> update our pallet's dependencies to include this:

Reviewer: ```suggestion
update our pallet's dependencies:
```
